msprime/trees.py

```diff
@@ -650,6 +676,12 @@ def draw(
         :param bool show_mutation_labels: If True, show labels for mutations.
         :param bool show_internal_node_labels: If True, show labels for internal nodes.
         :param bool show_leaf_node_labels: If True, show labels for leaf nodes.
+        :param list branch_colours: A list giving colours for edges in the tree.
+            If this is an array such as ['red','blue'], the indexes of the array
+            correspond to the node whose edges are coloured. If a dict, the keys
+            give the node numbers for each colour. Unspecified edges are coloured  
```

Comment:
We need to percolate colours up the tree, otherwise we may never spot coloured samples if they are on a small branch. My description is probably overly complicated, though. It's simply that if all children share the same colour, their parent should get that colour too (I think we need this functionality). If they don't all share the colour, we simply drop any bespoke colours for the parent node. That means the parents will take the default SVG colour for branches, which I have currently set to "dimgrey".

Agree about colour specs. They are fed straight into the SVG text at the moment, so anything that is acceptable to SVG is acceptable here. I don't do any sanity checking, so there is potential for SVG hacking, but since this is all currently client side, I'm not too worried.